pkg/storegateway/series_refs.go

```diff
@@ -800,13 +808,14 @@ func newLoadingSeriesChunkRefsSetIterator(
 	shard *sharding.ShardSelector,
 	seriesHasher seriesHasher,
 	skipChunks bool,
+	streamingSeries bool,
 	minTime int64,
 	maxTime int64,
 	tenantID string,
 	chunkRangesPerSeries int,
 	logger log.Logger,
 ) *loadingSeriesChunkRefsSetIterator {
-	if skipChunks {
+	if skipChunks && !streamingSeries {
 		minTime, maxTime = blockMeta.MinTime, blockMeta.MaxTime
 	}
```

Comment:
If we go with the enum value I think we will be able to remove these lines. I added them for explicitness, but an enum will do the same job imo